# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/base/strings/string16.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/base/strings/string16.cc
--- a/security/sandbox/chromium/base/strings/string16.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/base/strings/string16.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -15,7 +15,7 @@
 
 #include <ostream>
 
-#include "base/strings/utf_string_conversions.h"
+#include "base/strings/string_piece.h"
 
 namespace base {
 
@@ -47,7 +47,7 @@
     }
     ++s;
   }
-  return 0;
+  return nullptr;
 }
 
 char16* c16memmove(char16* s1, const char16* s2, size_t n) {
@@ -70,7 +70,7 @@
 namespace string16_internals {
 
 std::ostream& operator<<(std::ostream& out, const string16& str) {
-  return out << UTF16ToUTF8(str);
+  return out << base::StringPiece16(str);
 }
 
 void PrintTo(const string16& str, std::ostream* out) {